assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,58 @@
+( () => {
+	const tag = document.createElement( 'script' );
+	tag.id = 'iframe-api';
+	tag.src = 'https://www.youtube.com/iframe_api';
+	const firstScriptTag = document.getElementsByTagName( 'script' )[ 0 ];
+	firstScriptTag.parentNode.insertBefore( tag, firstScriptTag );
```

Comment:
Not sure what do you mean? Do you recommend replacing dynamic tag creation with text?
```
const youtubeScript = '<script src="https://www.youtube.com/iframe_api"></script>";
const firstScriptTag = document.getElementsByTagName( 'script' )[ 0 ];
firstScriptTag.parentNode.insertAdjacentHTML( 'beforebegin', youtubeScript );
```
It looks cleaner, yes :) Is it a preferable way?

Reply:
I think so! I'm just not sure if it will run properly the script. It's something good to test, and if it works, that seems better. Unless there's something in the YT recommendation recommending to use it in this way.

It also made me wonder if we really need to include it through JS, or if we could add it as dependency through WP when enqueueing our JS file. 🤔

Reply:
Unfortunately, it doesn't work this way: it does not evaluate the script.
However, the idea of registering and passing it as a dependency works well! Thanks!